File: Treachery-iOS/Treachery-iOS/Helpers/PlayerColors.swift
//
//  PlayerColors.swift
//  Treachery-iOS
//
//  Created by Luke Solomon on 3/18/26.
//

import SwiftUI

struct PlayerColor {
    let name: String
    let hex: String
    var color: Color { Color(hex: hex) }
}

enum PlayerColors {
    static let palette: [PlayerColor] = [
        PlayerColor(name: "Crimson", hex: "#e74c3c"),
        PlayerColor(name: "Sunset", hex: "#e67e22"),
        PlayerColor(name: "Amber", hex: "#f1c40f"),
        PlayerColor(name: "Emerald", hex: "#2ecc71"),
        PlayerColor(name: "Teal", hex: "#1abc9c"),
        PlayerColor(name: "Sky", hex: "#3498db"),
        PlayerColor(name: "Indigo", hex: "#6c5ce7"),
        PlayerColor(name: "Orchid", hex: "#a855f7"),
        PlayerColor(name: "Rose", hex: "#ec4899"),
        PlayerColor(name: "Silver", hex: "#95a5a6"),
    ]
}

File: TreacheryAndroid/app/src/main/java/com/solomon/treachery/ui/util/PlayerColors.kt
package com.solomon.treachery.ui.util

import androidx.compose.ui.graphics.Color

data class PlayerColor(
    val name: String,
    val hex: String
) {
    val color: Color
        get() {
            val cleaned = hex.removePrefix("#")
            return Color(android.graphics.Color.parseColor("#$cleaned"))
        }
}

object PlayerColors {
    val palette = listOf(
        PlayerColor("Crimson", "#e74c3c"),
        PlayerColor("Sunset", "#e67e22"),
        PlayerColor("Amber", "#f1c40f"),
        PlayerColor("Emerald", "#2ecc71"),
        PlayerColor("Teal", "#1abc9c"),
        PlayerColor("Sky", "#3498db"),
        PlayerColor("Indigo", "#6c5ce7"),
        PlayerColor("Orchid", "#a855f7"),
        PlayerColor("Rose", "#ec4899"),
        PlayerColor("Silver", "#95a5a6"),
    )
}
